### PR TITLE
Cleanup func-test suite

### LINKED ISSUE #3365: Review ignored tests in `func-test`
This is from a test-run for `func-test`:

```
haskell-language-server
  commands
    are prefixed:                                                          OK (32.02s)
    get de-prefixed:                                                       IGNORED
      Eval plugin disabled
  completions
    works:                                                                 OK (6.94s)
    itemCompletion/resolve works:                                          IGNORED
      no support for itemCompletion/resolve requests
    completes imports:                                                     OK (6.66s)
    completes qualified imports:                                           OK (6.91s)
    completes with no prefix:                                              OK (6.69s)
    recorddotsyntax
      shows field selectors:                                               OK (7.08s)
      shows field selectors for nested field:                              OK (6.69s)
    strips compiler generated stuff from completions:                      OK (6.56s)
    have implicit foralls on basic polymorphic types:                      OK (7.09s)
    have implicit foralls with multiple type variables:                    OK (7.05s)
    maxCompletions:                                                        OK (7.02s)
    import function completions:                                           OK (6.50s)
    import second function completion:                                     OK (6.72s)
    completes locally defined associated type family:                      OK (7.00s)
    contexts
      only provides type suggestions:                                      OK (6.93s)
      only provides value suggestions:                                     OK (6.56s)
      completes qualified type suggestions:                                OK (4.39s)
    snippets
      work for argumentless constructors:                                  FAIL
        Exception: Timed out waiting to receive a message from the server.
        Use -p '/work for argumentless constructors/' to rerun this test only.
      work for polymorphic types:                                          OK (4.64s)
      work for complex types:                                              OK (4.77s)
      work for infix functions:                                            OK (4.61s)
      work for infix functions in backticks:                               OK (4.62s)
      work for qualified infix functions:                                  OK (4.63s)
      work for qualified infix functions in backticks:                     OK (4.52s)
      respects lsp configuration:                                          OK (4.60s)
      respects client capabilities:                                        OK (4.62s)
      works for record fields sharing the single signature:                OK (4.55s)
  plugin config
    config parsing
      empty object as user configuration should not send error logMessage: OK (25.22s)
  deferred responses
    instantly respond to failed modules with no cache:                     OK (26.16s)
    multiple main modules
      Can load one file at a time, when more than one Main module exists:  IGNORED
        Broken: Unexpected ConduitParser.empty
  definitions
    goto's symbols:                                                        IGNORED
      Broken: file:///Users/jwindsor/src/haskell-language-server/test/testdata/References.hs
    goto's imported modules:                                               IGNORED
      Broken: file:///Users/jwindsor/src/haskell-language-server/test/testdata/Bar.hs
    goto's exported modules:                                               IGNORED
      Broken: file:///Users/jwindsor/src/haskell-language-server/test/testdata/Bar.hs
    goto's imported modules that are loaded:                               IGNORED
      Broken: file:///Users/jwindsor/src/haskell-language-server/test/testdata/Bar.hs
    goto's imported modules that are loaded, and then closed:              IGNORED
      Broken: file:///Users/jwindsor/src/haskell-language-server/test/testdata/Bar.hs
  diagnostics providers
    Warnings are warnings
      Overrides -Werror:                                                   OK (23.04s)
  format document
    works:                                                                 IGNORED
      Tests gets stuck in ci
    works with custom tab size:                                            IGNORED
      Tests gets stuck in ci
    format range
      works:                                                               IGNORED
        Tests gets stuck in ci
      works with custom tab size:                                          IGNORED
        Tests gets stuck in ci
    formatting provider
      respects none:                                                       IGNORED
        Tests gets stuck in ci
      can change on the fly:                                               IGNORED
        Tests gets stuck in ci
      supports both new and old configuration sections:                    IGNORED
        Tests gets stuck in ci
  behaviour on malformed projects
    no test executed:                                                      OK
  code actions
    import suggestions
      works with 3.8 code action kinds:                                    OK (27.89s)
    add package suggestions
      adds to .cabal files:                                                IGNORED
        no support for adding dependent packages via code action
      adds to hpack package.yaml files:                                    IGNORED
        no support for adding dependent packages via code action
    redundant import code actions
      remove solitary redundant imports:                                   OK (6.63s)
      doesn't touch other imports:                                         OK (6.34s)
    rename suggestions
      works:                                                               OK (26.49s)
      doesn't give both documentChanges and changes:                       OK (26.36s)
    missing top level signature code actions
      Adds top level signature:                                            OK (26.19s)
    typed hole code actions
      works:                                                               OK (26.21s)
      doesn't work when wingman is active:                                 FAIL (expected: The wingman plugin doesn't yet compile in GHC92) (26.35s)
        Exception: user error (didn't expected code action matching '["replace _ with minBound"]' but found one anyway) (expected failure)
      shows more suggestions:                                              OK (25.61s)
      doesnt show more suggestions when wingman is active:                 FAIL (expected: The wingman plugin doesn't yet compile in GHC92) (26.68s)
        Exception: user error (didn't expected code action matching '["replace _ with foo2 _"]' but found one anyway) (expected failure)
    unused term code actions
      Prefixes with '_':                                                   IGNORED
        no support for prefixing unused names with _
      respect 'only' parameter:                                            OK (22.36s)
  hie-bios
    loads modules inside main-is:                                          OK (1.06s)
    reports errors in hie.yaml:                                            FAIL (expected: hie-bios 0.11 has poor error messages) (0.88s)
      test/functional/HieBios.hs:33:
      Error reported (expected failure)
  highlight
    works:                                                                 OK (19.99s)
  window/workDoneProgress
    sends indefinite progress notifications:                               OK (20.16s)
    eval plugin sends progress reports:                                    IGNORED
      Eval plugin disabled
    ormolu plugin sends progress notifications:                            IGNORED
      Ormolu plugin disabled
    fourmolu plugin sends progress notifications:                          IGNORED
      Fourmolu plugin disabled
  references
    works with definitions:                                                IGNORED
      Broken
  document symbols
    pre 3.10 symbol information
      provides nested data types and constructors:                         OK (4.06s)
      provides nested where functions:                                     IGNORED
        extracting symbols from nested wheres not supported
      provides pattern synonyms:                                           IGNORED
        extracting pattern synonym symbols not supported
      provides imports:                                                    OK (3.68s)
    3.10 hierarchical document symbols
      provides nested data types and constructors:                         OK (3.42s)
      provides nested where functions:                                     IGNORED
        extracting symbols from nested wheres not supported
      provides pattern synonyms:                                           IGNORED
        extracting pattern synonym symbols not supported
      provides imports:                                                    OK (2.92s)
  type definitions
    finds local definition of record variable:                             OK (0.93s)
    finds local definition of newtype variable:                            OK (0.79s)
    finds local definition of sum type variable:                           OK (0.77s)
    finds local definition of sum type constructor:                        OK (0.75s)
    finds non-local definition of type def:                                OK (0.71s)
    find local definition of type def:                                     OK (0.64s)
    find type-definition of type def in component:                         OK (0.65s)
    find definition of parameterized data type:                            OK (0.62s)
```

Everything that is `IGNORED` should have a reason why it is ignored, deleted or fixed eventually.
Anyone can pick this up, just mark the test as not ignored, if it doesn't work, try to work out why it doesn't work, and update the description or try to fix it.

### PR DESCRIPTION
**Pull request by fendor**
This test-suite is vastly legacy from Haskell IDE Engine. It was originally written before we had a plugin-based architecture, where each plugin tests its feature in isolation.
Over time, this test-suite bitrotted, where a lot of testcases were either redundant or out-of-date and consequentially disabled.
Further, make sure we only have tests here that need to be integration tests.

We clean up the test-suite, delete old tests and remove unused testdata.

Before we merge:
* [x] Move tests for qualified imports to the plugin the feature is defined in
* [x] Write a README for guidance when to add a test to `func-test`.

closes #3365

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -3,7 +3,8 @@
 /ghcide/session-loader @pepeiborra @fendor
 /hls-graph @pepeiborra
 /hls-plugin-api @berberman
-/hls-test-utils
+/hls-test-utils @fendor
+/test @fendor
```
Comment (fendor) [inline CODEOWNERS:7]: I noticed that `hls-test-utils` has no codeowner, I am volunteering to take over the ownership.
Additionally, added myself as codeowner for `func-test`. Mostly, to make sure my somewhat arbitrary guidance in `test/README.md` cannot be further violated without me noticing.

**Review by michaelpj — APPROVED** [review]
FunctionalBadProject and HieBios also looks like they could be in ghcide?

**Comment by fendor** [comment]
Yeah, that's true.

**Comment by fendor** [comment]
I think, the only tests that should really be in `func-test` are the formatter provider tests, and the config change tests.

The rest could live somewhere else...

**Comment by fendor** [comment]
For some reason, the tests get slower if I move them to ghcide. Until I got to the bottom of this, I won't move the tests, and leave them here.
Merge as-is to speed up CI a little bit.

**Pull request merged**